index.js

```diff
@@ -6,13 +6,153 @@ function ask(questionText) {
     rl.question(questionText, resolve);
   });
 }
+//Initialize a counter for the number of guesses, as well as a max number and min number for the range.
+let guessCount = 1
+let maxNum = 100
+let minNum = 1
+//Initialize a random number using binary search with the min and max values.
+let randNum = Math.floor((maxNum + minNum) / 2)
 
-start();
-
+//function that asks the user to guess a randomly generated number by the computer.
+async function reverseGame() {
+  //Initialize a different random number for this specific function. This number is generated randomly from 1-100.
+  let randNum = Math.floor(Math.random() * (maxNum - minNum + 1)) + minNum
```

Comment:
you have already initialized randNum as a global variable on line 14 as a random num generator. this initialization is local only to your reverseGame function, and contains the logic for a smart guessing random num generator.